packages/transformer/src/IModelImporter.ts

```diff
@@ -597,9 +620,28 @@ export class IModelImporter implements Required<IModelImportOptions> {
       options: this.options,
       targetDbId: this.targetDb.iModelId || this.targetDb.nativeDb.getFilePath(),
       doNotUpdateElementIds: CompressedId64Set.compressSet(this.doNotUpdateElementIds),
+      duplicateCodeValueMap: Object.fromEntries(this._duplicateCodeValueMap),
       additionalState: this.getAdditionalStateJson(),
     };
   }
+
+  private resolveDuplicateCodeValues(): void {
+    for (const [elementId, codeValue] of this._duplicateCodeValueMap) {
+      const element = this.targetDb.elements.getElement(elementId);
+      element.code.value = codeValue;
+      element.update();
+    }
+  }
+
+  /**
+   * Needs to be called to perform necessary cleanup operations.
+   * By not calling `finalize` there is a risk that data imported into targetDb will not be as expected.
+   *
+   * @internal
```

Comment:
This should probably not be internal... users of just `IModelImporter` will need it. Maybe note that this doesn't need to be called if using the transformer to manager the importer.

Reply:
removed internal attribute.
Also now clearing the map in `resolveDuplicateCodeValues` , so re-invoking the method will effectvely be a noop.